### PR TITLE
WT-9242 Process the call log for prepare_transaction

### PR DESCRIPTION
**Pull request by Siddhartha8899**

### CODE REVIEW

File: test/simulator/timestamp/src/session_simulator.cpp [hunk 1]
```diff
@@ -111,6 +111,31 @@ session_simulator::rollback_transaction(const std::string &config)
...
+    timestamp_manager *ts_manager = &timestamp_manager::get_timestamp_manager();
+    std::map<std::string, std::string> config_map;
+
+    const std::vector<std::string> supported_ops = {"prepare_timestamp"};
+    const std::vector<std::string> unsupported_ops;
+
+    WT_SIM_RET_MSG(ts_manager->parse_config(config, supported_ops, unsupported_ops, config_map),
+      "Incorrect config (" + config + ") passed in prepare_transaction");
+
+    auto pos = config_map.find("prepare_timestamp");
```
Comment (bvpvamsikrishna) [inline test/simulator/timestamp/src/session_simulator.cpp:130]: Are there no rules for the prepare timestamp?

Comment (Siddhartha8899) [reply test/simulator/timestamp/src/session_simulator.cpp:130]: This will be done in WT-9232, I will add a fix me comment for now.

Comment (Siddhartha8899) [reply test/simulator/timestamp/src/session_simulator.cpp:135]: Rules for the prepared transaction is added to this PR, however, as mentioned earlier that the rules for prepared timestamp will be added in WT-9232

File: test/simulator/timestamp/src/session_simulator.cpp [hunk 2]
```diff
@@ -129,8 +134,21 @@ session_simulator::prepare_transaction(const std::string &config)
...
+        WT_SIM_RET(ts_manager->validate_hex_value(pos->second, "prepare timestamp"));
+        uint64_t prepare_ts = ts_manager->hex_to_decimal(pos->second);
+        set_prepare_timestamp(prepare_ts);
     }
 
+    /* A prepared timestamp should have been set at this point. */
+    if (!has_prepare_timestamp())
+        WT_SIM_RET_MSG(EINVAL, "'prepare_transaction' - prepare timestamp is not set");
+
+    /* commit timestamp must not be set before transaction is prepared. */
```
Comment (etienneptl) [inline test/simulator/timestamp/src/session_simulator.cpp:146]: `commit` -> `Commit`

**Review by etienneptl — APPROVED** [review]
LGTM after one np

File: test/simulator/timestamp/src/session_simulator.cpp [hunk 3]
```diff
@@ -111,6 +111,49 @@ session_simulator::rollback_transaction(const std::string &config)
...
+        uint64_t prepare_ts = ts_manager->hex_to_decimal(pos->second);
+        set_prepare_timestamp(prepare_ts);
+    }
+
+    /* A prepared timestamp should have been set at this point. */
+    if (!has_prepare_timestamp())
+        WT_SIM_RET_MSG(EINVAL, "'prepare_transaction' - prepare timestamp is not set");
+
+    /* commit timestamp must not be set before transaction is prepared. */
+    if (_has_commit_ts)
```
Comment (bvpvamsikrishna) [inline test/simulator/timestamp/src/session_simulator.cpp:147]: could we move this check to line 125, that way the validity of the transaction is verified before processing any config?

**Review by bvpvamsikrishna — APPROVED** [review]
lgtm, after the comment about moving the commit ts check to start of the function.

File: test/simulator/timestamp/src/session_simulator.cpp (hunk 2 above)
Comment (Siddhartha8899) [reply test/simulator/timestamp/src/session_simulator.cpp]: Updated in [e3e29bb](https://github.com/wiredtiger/wiredtiger/pull/8386/commits/e3e29bba3019a6429b3fbb5835e83da23c9cc41b)

File: test/simulator/timestamp/src/session_simulator.cpp (hunk 3 above)
Comment (Siddhartha8899) [reply test/simulator/timestamp/src/session_simulator.cpp:147]: The code flow is similar to wiredtiger such that we get the same errors. Let me know what you think.

Comment (Siddhartha8899) [reply test/simulator/timestamp/src/session_simulator.cpp:147]: I will merge this ticket, if this remains a concern to you then we can look this in the next PR for WT-9232

**Pull request merged**